util/ranger/types.go

```diff
@@ -134,12 +134,29 @@ func (ran *NewRange) PrefixEqualLen(sc *stmtctx.StatementContext) (int, error) {
 	return len(ran.LowVal), nil
 }
 
-func formatDatum(d types.Datum) string {
-	if d.Kind() == types.KindMinNotNull {
+func formatDatum(d types.Datum, isLeftSide bool) string {
+	switch d.Kind() {
+	case types.KindNull:
+		return "<nil>"
+	case types.KindMinNotNull:
 		return "-inf"
-	}
-	if d.Kind() == types.KindMaxValue {
+	case types.KindMaxValue:
 		return "+inf"
+	case types.KindInt64:
+		switch d.GetInt64() {
+		case math.MinInt64:
+			if isLeftSide {
+				return "-inf"
+			}
+		case math.MaxInt64:
+			if !isLeftSide {
+				return "+inf"
+			}
+		}
+	case types.KindUint64:
+		if d.GetUint64() == math.MaxUint64 && !isLeftSide {
```

Comment:
When is it MaxUint64 but is on the left side.

Reply:
`a >= MaxUint64` the actual range will be `[MaxUint64, MaxUint64]`. The best result to display is `[MaxUint64, +inf]` i think